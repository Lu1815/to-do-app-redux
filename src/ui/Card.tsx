import React from 'react'
import HeaderOne from './HeaderOne'

interface ICardProps {
    text?: string
}

const Card = ({text}: ICardProps) => {
  return (
    <div className='w-full py-2 px-3 border-gray-200 border-2 rounded-md'>
        <HeaderOne content={text ?? 'Card Text'} />
    </div>
  )
}

export default Card