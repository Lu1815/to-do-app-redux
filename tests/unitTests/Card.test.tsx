import React from 'react';
import { render, screen } from '@testing-library/react';
import Card from '../../src/ui/Card';

describe('Card Component', () => {
  it('renders without crashing', () => {
    render(<Card text="Test Card" />);
    expect(screen.getByText('Test Card')).toBeInTheDocument();
  });

  it('renders with default text when text prop is not provided', () => {
    render(<Card />);
    expect(screen.getByText('Card Text')).toBeInTheDocument();
  });

  it('matches snapshot', () => {
    const { container } = render(<Card text="Snapshot Test" />);
    expect(container).toMatchSnapshot();
  });

  it('applies proper styling', () => {
    render(<Card text="Styling Test" />);
    const cardElement = screen.getByText('Styling Test').parentElement;

    expect(cardElement).toHaveClass('w-full');
    expect(cardElement).toHaveClass('py-2');
    expect(cardElement).toHaveClass('px-3');
    expect(cardElement).toHaveClass('border-gray-200');
    expect(cardElement).toHaveClass('border-2');
    expect(cardElement).toHaveClass('rounded-md');
  });

  it('renders with custom text', () => {
    render(<Card text="Custom Text" />);
    expect(screen.getByText('Custom Text')).toBeInTheDocument();
  });
});
