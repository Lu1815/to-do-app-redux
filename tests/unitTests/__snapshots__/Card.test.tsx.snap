// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<div>
  <div
    class="w-full py-2 px-3 border-gray-200 border-2 rounded-md"
  >
    <p
      class="text-lg text-black font-normal"
    >
      Snapshot Test
    </p>
  </div>
</div>
`;
